endpoints/openrtb2/amp_auction.go

```diff
@@ -215,12 +225,13 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		StoredBidResponses:         storedBidResponses,
 		BidderImpReplaceImpID:      bidderImpReplaceImp,
 		PubID:                      labels.PubID,
+		HookExecutor:               deps.hookExecutor,
 	}
 
 	response, err := deps.ex.HoldAuction(ctx, auctionRequest, nil)
 	ao.AuctionResponse = response
-
-	if err != nil {
+	reject := hookexecution.FindReject([]error{err})
+	if err != nil && reject == nil {
```

Comment:
The `FindReject` function is allocating memory just to check the type, which is not ideal for performance. I also don't believe it needs to return a pointer. Please implement something like a new `CastRejectErr` method:

```
func FindFirstRejectOrNil(errors []error) RejectError {
	for _, err := range errors {
		if rejectErr, ok := CastRejectErr(err); ok {
			return rejectErr
		}
	}
	return nil
}

func CastRejectErr(err error) (RejectError, bool) {
    return err.(RejectError)
}
```

and update the call side to:

```
rejectErr := hookexecution.CastRejectErr(err)
```

Reply:
Sure, we will add `CastRejectErr` func to avoid extra memory allocation. However, since we work with the concrete type `RejectError`, we are forced to return a pointer if we want to return `nil` when `RejectError` is not found.

Reply:
Ah, ok. Got it. Makes sense.